planner/core/rule_aggregation_elimination.go

```diff
@@ -179,25 +179,25 @@ func wrapCastFunction(ctx sessionctx.Context, arg expression.Expression, targetT
 	return expression.BuildCastFunction(ctx, arg, targetTp)
 }
 
-func (a *aggregationEliminator) optimize(ctx context.Context, p LogicalPlan) (LogicalPlan, error) {
+func (a *aggregationEliminator) optimize(ctx context.Context, p LogicalPlan, opts ...logicalOptimizeOption) (*logicalOptimizeReturn, error) {
 	newChildren := make([]LogicalPlan, 0, len(p.Children()))
 	for _, child := range p.Children() {
 		newChild, err := a.optimize(ctx, child)
```

Comment:
Do we need to add `opts` here?

Reply:
currently we only implement framework for rule.optimize. In future we will implement specific logical code for each rule during optimize. Thus no need to add `opts` here for now.